internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,207 @@
+		}
+		root, err := gopache.Parse(confFile)
+		if err != nil {
+			logrus.Errorf("Error while parsing configuration file : %s", err)
+			return port, confFileName, err
+		}
+		match, err := root.FindOne(virtualHost)
+		if err != nil {
+			logrus.Errorf("Could not find the VirtualHost in conf file: %s", err)
+			return port, confFileName, err
+		}
+		tokens := strings.Split(match.Content, ":")
+		if len(tokens) > 1 {
+			port, err := strconv.ParseInt(tokens[1], 10, 32)
+			if err != nil {
+				logrus.Errorf("Error while converting the entered port from string to int : %s", err)
+				return int32(port), confFileName, err
+			}
+			return int32(port), confFileName, nil
+		}
+		defer confFile.Close()
+	}
+	return port, confFileName, err
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PHPDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return nil, nil, err
+	}
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
+		if ext == phpExt {
+			unnamedServices = []transformertypes.TransformerPlan{{
+				Mode:              t.Config.Spec.Mode,
+				ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+				BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+				},
+			}}
+			return nil, unnamedServices, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PHPDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var detectedPorts []int32
+		var phpConfig PhpTemplateConfig
+		port, confFileName, err := DetectConf(a.Paths[artifacts.ProjectPathPathType][0])
```

Comment:
I assume the conf file can be in any sub directory, so we should be checking all sub directories.

Reply:
Yes, it's possible that the conf file may be present in a sub-directory. So, I will check for it in all sub directories.

Reply:
You can possibly use `common.GetFilesByExt` function.

Reply:
Yes, I am making the required changes using that. It might be possible that there may be more than one conf file as well. Then, we should copy all of them.

Reply:
If there is more than one, we should ask the user which one to use

Reply:
okay sure